internal/service/alarms/internal/alertmanager/alertmanager.go

```diff
@@ -61,3 +67,104 @@ func Setup(ctx context.Context, cl client.Client) error {
 	slog.Info("Successfully configured alertmanager")
 	return nil
 }
+
+// ConvertAmToAlarmEventRecordModels get alarmEventRecords based on the alertmanager notification and AlarmDefinition
+func ConvertAmToAlarmEventRecordModels(am *api.AlertmanagerNotification, aDefinitionRecords []models.AlarmDefinition) []models.AlarmEventRecord {
+	var records []models.AlarmEventRecord
+
+	for _, alert := range am.Alerts {
+		slog.Info("Converting Alertmanager alert", "alert name", GetAlertName(*alert.Labels))
+		record := models.AlarmEventRecord{
+			AlarmRaisedTime:       *alert.StartsAt, // nothing for Changed time
+			AlarmClearedTime:      alert.EndsAt,
+			PerceivedSeverity:     getPerceivedSeverity(*alert.Labels),
+			NotificationEventType: 0,
+			AlarmStatus:           string(*alert.Status),
+			Fingerprint:           *alert.Fingerprint,
+			ResourceID:            GetResourceID(*alert.Labels),
+		}
+
+		// Update Extensions with things we didn't really process
+		record.Extensions = getExtensions(*alert.Labels, *alert.Annotations)
+
+		// See if possible to pick up additional info from its definition
+		for _, def := range aDefinitionRecords {
+			if def.AlarmName == GetAlertName(*alert.Labels) { // TODO: match with the resourceTypeID too once init data is ready
+				record.AlarmDefinitionID = def.AlarmDefinitionID
+				record.ProbableCauseID = def.ProbableCauseID
+			}
+		}
+
+		records = append(records, record)
+	}
+
+	return records
+}
+
+// GetResourceID for caas alerts it's the cluster ID
+func GetResourceID(labels map[string]string) uuid.UUID {
+	val, ok := labels["managed_cluster"]
+	if !ok {
+		slog.Warn("Could not find managed_cluster", "alertname", GetAlertName(labels))
+		return uuid.UUID{}
+	}
+
+	id, err := uuid.Parse(val)
+	if err != nil {
+		slog.Warn("Could convert managed_cluster string too uuid", "err", err.Error())
+		return uuid.UUID{}
+	}
+
+	return id
+}
+
+// GetResourceTypeID get resource Type ID.
+// TODO Update this so that resourceTypeID correctly mapped to resourceID
+func GetResourceTypeID(labels map[string]string) uuid.UUID {
+	return GetResourceID(labels)
+}
+
+// GetAlertName extract name from alert label
+func GetAlertName(labels map[string]string) string {
+	val, ok := labels["alertname"]
+	if !ok {
+		// this may never execute but keeping a check just in case
+		return "Unknown"
+	}
+
+	return val
+}
+
+// getPerceivedSeverity am's `severity` to oran's PerceivedSeverity
+func getPerceivedSeverity(labels map[string]string) api.PerceivedSeverity {
+	input, ok := labels["severity"]
+	if !ok {
+		slog.Warn("Could not find severity label. This may impact the value of PerceivedSeverity")
```

Comment:
maybe dump out the other fields so we can track down which alert is causing this.

Reply:
added. printing all the labels now.